src/App.tsx

```diff
@@ -1,14 +1,71 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+// eslint-disable-next-line import/extensions
+import { FilterCriteria, Select, Status } from './types/otherTypes';
+
+const filterTodos = (
+  todos: Todo[],
+  { status, searchInput }: FilterCriteria,
+): Todo[] => {
+  return todos.filter(todo => {
+    if (status === Status.Active && todo.completed) {
+      return false;
+    }
+
+    if (status === Status.Completed && !todo.completed) {
+      return false;
+    }
+
+    if (
+      searchInput &&
+      !todo.title.toLowerCase().includes(searchInput.toLowerCase())
+    ) {
+      return false;
+    }
+
+    return true;
+  });
+};
 
 export const App: React.FC = () => {
+  const [isTodosLoading, setIsTodosLoading] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Select>(null);
+  const [todosFromServer, setTodosFromServer] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<FilterCriteria>({
+    status: Status.All,
+    searchInput: '',
+  });
+
+  useEffect(() => {
+    setIsTodosLoading(true);
+    getTodos()
+      .then((todos: Todo[]) => {
+        const filteredTodos = filterTodos(todos, filter);
+
+        setTodosFromServer(filteredTodos);
+      })
+      .finally(() => {
+        setIsTodosLoading(false);
+      });
+  }, [filter]);
```

Comment:
You don't need to load todos from the server every time the filter changes. Do it just once